realsense_camera/include/realsense_camera/base_nodelet.h

```diff
@@ -173,6 +173,8 @@ class BaseNodelet: public nodelet::Nodelet
   virtual void publishDynamicTransforms();
   virtual void prepareTransforms();
   virtual void checkError();
+  virtual void checkFirmwareValidation(std::string validatedFirmware,
+        rs_device* rs_detected_device, std::string typeOfFirmware);
```

Comment:
Please change rs_detected_device to const rs_device& instead of pointer. See this article for explanation: http://www.cplusplus.com/articles/z6vU7k9E/